internal/request/adapters/http_client.go

```diff
@@ -31,6 +35,16 @@ func (c *httpClient) Do(req *domain.Request) (*domain.Response, error) {
 		return nil, err
 	}
 
+	if req.Headers != nil {
+		for key, element := range req.Headers {
+			if strings.ToLower(key) == "host" {
+				httpReq.Host = element
```

Comment:
For whatever reason, the golang client does not accept the `Host` as header, we need to set it here on this property.

Reply:
Nice one!